Download page structural and stylistic overhaul

patrickbkr opened the pull request:
This PR is a proposal of a rather large overhaul of the rakudo.org website.
A deployment of the website is available at <https://stage-rakudo.rakulang.site/>.

As there are no precomp archives uploaded yet, there are no files shown on the files page -> Windows/Mac OS/Linux tabs -> `Archives` and `Installer` sections. Once we actually upload archives and installers files will show up there, similar to the files list on the `Sources` tab.

Following is a list of things I changed (in no particular order):

- "Rakudo Star" consistently called "Rakudo Star Bundle" everywhere, to communicate right in the name, that it's a software bundle. This should counter the confusion of Raku vs Rakudo vs Rakudo Star for newcomers a bit.
- Rakudo Star Bundle deemphasized. It was once marketed as the primary thing to download. I think now that we can provide smaller, Rakudo only, precompiled archives we shouldn't emphasize the Star Bundle as much on the rakudo.org website (given it's a bundle of all sorts of modules).
- Added precompiled releases prominently.
- Added release files JSON API at <https://stage-rakudo.rakulang.site/dl/rakudo> that provides a machine readable index of all the release files there are.
- Switch to a different file naming scheme that also encodes OS, architecture and a build revision. File names of source tarballs are backwards compatible and stay unchanged. Star Bundle release files will need to be named with the help of a script ([rename-files.p6](https://gist.github.com/patrickbkr/53a7494d228888d466386448d30eb16e), still a bit WIP)).
- The forwarding of the release listings to GitHub should be removed. The change to a different website when clicking a download button is quite disconcerting.
- Many stylistic changes: some icons changed, be more bold in the start page description, file page completely restructured, color scheme slightly changed.

The new downloads page strongly promotes the precompiled archives and installers. For this to make much sense we do need to provide such archives. I can create a set (Windows, MacOS, Linux) for the 2019.11 release. We do need to make sure we are releasing such archives regularly though. @nxadm is working on automating that (using Travis), but is not done yet. Installers for Windows and Mac OS would be nice to have, but I think nobody is actively working on that atm.

@pmurias, @usev6: I think you two have maintained the JVM backend in the last years mostly. The JVM backend could be featured more on the files page but isn't because my perception is, that the JVM backend is rather WIP at the moment and encouraging people to just use it is not what we want to do. Is that in your best interest?
@pmurias: You recently wanted more feedback on the JavaScript backend. The files page is prominently listing JavaScript as a backend and links to the npm page. Is this in your interest?
@AlexDaniel, @Altai-man: There was some concern removing the forward to GitHub might break stuff. How come and what can we do about that?

usev6 commented:
@patrickbkr I fully agree that we shouldn't feature the JVM backend prominently. It's experimental/WIP and definitely not recommended for production.

nxadm commented:
I am not fond of "Archives" because it has the connotation of "archived releases", i.e. old files. In the case of Linux I propose to use "Binary releases" with a short note they are relocatable. I agree that this would need to be the first entry.

You'll find the ones for 2019.11 [here](https://github.com/nxadm/rakudo-pkg/releases/tag/v2019.11). Their creation is automated on Travis.

patrickbkr commented:
Several people have now argued for a different naming of the sub pages. So I renamed stuff extensively.
I also changed the heading of the precompiled packages.
@nxadm: Did you close this PR on purpose?

abraxxa commented:
Can you please update https://stage-rakudo.rakulang.site with the committed changes? Thanks!

abraxxa commented:
I like the current, brights color schema more, especially the download page but everything should be a bit more colorful.

nxadm commented:
@patrickbkr, oh, my excuses, The closing was not intended.

patrickbkr commented:
(The above is just a rebase.)

I have created binary builds for Linux, MacOS and Windows for the Rakudo 2019.11 release. They can be downloaded here: [https://patszim.volans.uberspace.de/patcloud/index.php/s/qmkreEJLQDZcjbf](https://patszim.volans.uberspace.de/patcloud/index.php/s/qmkreEJLQDZcjbf)
They should be placed in the same folder where the source release files are on the rakudo.org server.
AFAIK That's the last missing bit that was blocking a merge.
I'd like to get this deployed if there are no blockers as my work on rakubrew is now mostly blocked by this merge.

@rba What's your opinion?

rba commented:
@patrickbkr Sorry for letting you wait.
I've added the files to the downloads/rakudo folder and they are visible now at https://stage-rakudo.rakulang.site/.

Yet the download is not working yet due to not yet adjusted http rewrite rules. We have to fix the nginx config used on the prod server. Currently used: https://gist.github.com/3f5a52f5bde947adec3199cf8ed7e932

After this, I would merge this PR if there is no additional feedback.

Altai-man commented:
Very cool to see the update! Can you also address:
>Also, community page does mention #raku-dev channel, but not #raku, we should promote it more.

?

rba commented:
> Very cool to see the update! Can you also address:
> 
> > Also, community page does mention #raku-dev channel, but not #raku, we should promote it more.
> 
> ?

In the footer of every page is #raku and on the #raku-dev mentioned too.
